scripts/postinstall.sh

```diff
@@ -138,51 +149,72 @@ add_prefixes_to_npmrc() {
 do_check_and_install_npmrc() {
     local NPMRC_FILE="$1"
     ensure_npmrc_exists "$NPMRC_FILE"
-    
+
     find_missing_prefixes "$NPMRC_FILE"
-    
+
     if [[ "${#MISSING_PREFIXES[@]}" -eq 0 ]]; then
         echo "No missing prefixes in $NPMRC_FILE."
-        
+
         if ! $FORCE; then
             echo "Use --force (-f) to override and update the file"
         else
             backup_file "$NPMRC_FILE"
         fi
     fi
-    
+
     add_prefixes_to_npmrc "$NPMRC_FILE" "${MISSING_PREFIXES[@]}"
 }
 
 # Function to start the DPM registry if not running
 start_dpm_registry() {
-    REGISTRY_PID="$(cat $REGISTRY_PID_FILE 2>/dev/null || pgrep -f $REGISTRY_PROCESS_NAME || ps aux | grep $REGISTRY_PROCESS_NAME | awk '{print $2}' || lsof -i :2092 | grep node | awk '{print $2}')"
-    if [[ -z "$REGISTRY_PID" ]]; then
-        echo "Starting registry ..."
-        sh scripts/registry.nohup.sh
+    if docker version >/dev/null 2>&1; then
+        sh ./scripts/registry.docker.sh
     else
-        read -p "Registry running, (pid=$REGISTRY_PID). Would you like to restart the registry process? [y/N] " ANSWER
-        if [[ "$ANSWER" =~ ^[yY]$ ]]; then
-            echo "Restarting registry process (pid=$REGISTRY_PID) ..."
-            kill -9 "$REGISTRY_PID"
-            sh scripts/registry.nohup.sh
-            echo "Registry restarted (pid=$(pgrep -f \"$REGISTRY_PROCESS_NAME\"))"
+        # shellcheck disable=SC2009
+        REGISTRY_PID="$(cat $REGISTRY_PID_FILE 2>/dev/null || \
+            pgrep -f $REGISTRY_PROCESS_NAME \
+            || ps aux | grep $REGISTRY_PROCESS_NAME | grep -v grep | awk '{print $2}' \
+            || lsof -i :2092 | grep node | awk '{print $2}')"
+
+        if [[ -z "$REGISTRY_PID" ]]; then
+            roomy_echo "Starting registry ..."
+            sh ./scripts/registry.nohup.sh
         else
-            echo "Registry still running, continuing ..."
+            echo ""
+            read -rp "Registry running, (pid=$REGISTRY_PID). Would you like to restart the registry process? [y/N] " ANSWER
+            if [[ "$ANSWER" =~ ^[yY]$ ]]; then
+                echo "Restarting registry process (pid=$REGISTRY_PID) ..."
+                kill -9 "$REGISTRY_PID"
+                sh scripts/registry.nohup.sh
+                echo "Registry restarted (pid=$(pgrep -f \"$REGISTRY_PROCESS_NAME\"))"
+            else
+                echo "Registry still running, continuing ..."
+            fi
         fi
     fi
-    edit_hosts
 }
 
+# NOTE: I really don't think we should be touching people's /etc/hosts file.
+# Perhaps suggest it to them, but that's pretty "invasive" IMO
```

Comment:
Yeah I agree. I did hours of research to find a way to map 127.0.0.1 to a custom domain without doing this, but didn't find anything. if you have suggestions, let me know.